### ISSUE TITLE
CodeEditor - Un-indent code shortcut

### ISSUE THREAD

**Issue by kelvah** [patternfly/patternfly-react#8463]
**Is this a new component or an extension of an existing one?**
This is a new feature for the existing `CodeEditor` component

**Describe the feature**
I've noticed that the `CodeEditor` component [adds a command](https://github.com/patternfly/patternfly-react/blob/fbbc405bd078b4be09afc735f06ea5e924a15870/packages/react-code-editor/src/components/CodeEditor/CodeEditor.tsx#L384) for the `shift-tab` shortcut that focuses the monaco-editor parent element.

We're going to use `CodeEditor` for YAML code in our project. Indentation is important for YAML because it's how it denotes nesting. Usually `shift-tab` is the common shortcut used within IDEs to un-indent code. 

I was wondering if it could make sense to let users configure the `CodeEditor` to not override the default `shift-tab` behavior.

It's still possible to remove indentation by deleting tabs manually but I'm afraid that not being able to use `shift-tab` could be annoying in the long run.

**Are there visuals for this feature? If applicable, please include examples for each state and for varying widths**
Not applicable

**Comment by nicolethoen** [comment]
We had to brainstorm for a while how to enable keyboard users to navigate in and out of the code editor. 
By design, the [monaco code editor traps focus within it](https://github.com/microsoft/monaco-editor/wiki/Monaco-Editor-Accessibility-Guide#tab-trapping). Monaco's built in keyboard interaction is documented to be `You can toggle the trapping of Tab with Ctrl+M on Windows and Linux and with Ctrl+Shift+M on OSX, and subsequent Tab keys will move focus out of the editor.` The PatternFly developers found this to be unintuitive and incredibly buggy. We do have a feature allowing the code editor to display helpful keyboard shortcuts, but a keyboard user can only view these shortcuts, if they can navigate out of the code editor via keyboard. So an unintuitive pattern is not particularly helpful.

Being sure that the code editor is not trapping keyboard focus is a high priority accessibility concern. It had been flagged as a violation in [this VPAT (page 4 item 2.1.2)](https://access.redhat.com/sites/default/files/attachments/openshift-4-vpat.pdf).

Another factor for the decision being the way it is now is because it is a standard keyboard interaction used in text editors that `command/control` + `[`|`]` can be used to indent or un-indent text, so we have not overridden the only keyboard interaction.

All that being said, I did spend some time brainstorming a few other options. Another standard keyboard interaction to exit or close a component is to use the `esc` key. I fear that overriding the escape key will cause even more headaches for users of the code editor.

In the meantime, you can override our override by adding a line to the editorDidMount function. Something like:
```
 editor.addCommand(monaco.KeyMod.Shift | monaco.KeyCode.Tab, () => //backspace x number spaces);

```
But if you do that, you should still be sure to accommodate the focus trapping in another intuitive way.

@jessiehuff @thatblindgeye @srambach

**Comment by thatblindgeye** [comment]
Based on the comment above and the inactivity I'll be closing this. Please let us know if the workaround/comment above does not suffice or if you are still having any issues.
